lib/matplotlib/category.py

```diff
@@ -4,15 +4,33 @@
 """
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
-
 import six
 
 import numpy as np
 
-import matplotlib.cbook as cbook
 import matplotlib.units as units
 import matplotlib.ticker as ticker
 
+# np 1.6/1.7 support
+from distutils.version import LooseVersion
+import collections
+
+
+def shim_array(data):
```

Comment:
might be better to do this as

```python
if LooseVersion(np.__version__) < LooseVersion('1.8.0'):
   def shim_array(data):
        ...
else:
    def shim_array(data):
        return np.array(data, dtype=np.unicode)
```

so we only do the version check at import time.